### ISSUE TITLE
TemporalFusionTransformer does not perform well as expect.

### ISSUE THREAD

**Issue by leon90dm** [jdb78/pytorch-forecasting#675]
- PyTorch-Forecasting version: 0.9.0
- PyTorch version:1.9.0+cu102
- Python version:3.6
- Operating System:Centos

### Expected behavior
I just created a fool dataset, with four columns like below:

row_no | groups | time_idx | close
0 | 0 | 1 | 4
1 | 0 | 2 | 5
2 | 0 | 3 | 6
3 | 0 | 4 | 7
4 | 0 | 5 | 8

where `close = time_idx + 3`, Using 14 steps information to predict 14 steps afterwards. I expect TemporalFusionTransformer will easily catch the relation between time_idx and close.

### Actual behavior

Train loss is quite low, but valid loss is high. And with time_idx grows the prediction become quite different from true close.

### Code to reproduce the problem

Data Creation:

```
data = pd.DataFrame({
    'row_no':np.arange(1400)
})
data['groups'] = (data.row_no/28).astype(int)

data.groups.value_counts().shape

data['time_idx'] = data.groupby('groups').apply(lambda x: x['row_no']-x['row_no'].min() + 1).droplevel(0)

data['close'] = data.time_idx + 3
```

```
max_prediction_length = 14
max_encoder_length = 14
training_cutoff = data["time_idx"].max() - max_prediction_length

training = TimeSeriesDataSet(
        data[data.time_idx <= training_cutoff],
        time_idx="time_idx",
        target="close",
        group_ids=['groups'],
        min_encoder_length=7,  # keep encoder length long (as it is in the validation set)
        max_encoder_length=14,
        min_prediction_length=7,
        max_prediction_length=max_prediction_length,
        static_categoricals=[],
        static_reals=[],
        time_varying_known_categoricals=[],
        variable_groups={},  # group of categorical variables can be treated as one variable
        time_varying_known_reals=['time_idx'],
        time_varying_unknown_categoricals=[],
        time_varying_unknown_reals=[],
        target_normalizer=GroupNormalizer(groups=['groups']),  # use softplus and normalize by group
        add_relative_time_idx=False,
        add_target_scales=False,
        add_encoder_length=False,
    )
```
Training Dataset Samples:

```
({'encoder_cat': tensor([], size=(1, 7, 0), dtype=torch.int64),
  'encoder_cont': tensor([[[-1.6125],
           [-1.3644],
           [-1.1163],
           [-0.8682],
           [-0.6202],
           [-0.3721],
           [-0.1240]]]),
  'encoder_target': tensor([[ 4.,  5.,  6.,  7.,  8.,  9., 10.]]),
  'encoder_lengths': tensor([7]),
  'decoder_cat': tensor([], size=(1, 7, 0), dtype=torch.int64),
  'decoder_cont': tensor([[[0.1240],
           [0.3721],
           [0.6202],
           [0.8682],
           [1.1163],
           [1.3644],
           [1.6125]]]),
  'decoder_target': tensor([[11., 12., 13., 14., 15., 16., 17.]]),
  'decoder_lengths': tensor([7]),
  'decoder_time_idx': tensor([[ 8,  9, 10, 11, 12, 13, 14]]),
  'groups': tensor([[0]]),
  'target_scale': tensor([[10.5000,  4.1833]])},
 (tensor([[11., 12., 13., 14., 15., 16., 17.]]), None))
```

DataLoader:

```
validation = TimeSeriesDataSet.from_dataset(training, data, predict=True, stop_randomization=True)

# create dataloaders for model
batch_size = 1 # set this between 32 to 128
train_dataloader = training.to_dataloader(train=True, batch_size=batch_size, num_workers=2)
val_dataloader = validation.to_dataloader(train=False, batch_size=batch_size, num_workers=2)
```

Model:

```
early_stop_callback = EarlyStopping(monitor="val_loss", min_delta=1e-4, patience=3, verbose=False, mode="min")
lr_logger = LearningRateMonitor()  # log the learning rate
logger = TensorBoardLogger("lightning_logs")  # logging results to a tensorboard

trainer = pl.Trainer(
    max_epochs=20,
    gpus=0,
    weights_summary="top",
    gradient_clip_val=0.1,
    limit_train_batches=8,  # coment in for training, running valiation every 30 batches
    # fast_dev_run=True,  # comment in to check that networkor dataset has no serious bugs
    callbacks=[lr_logger, early_stop_callback],
    logger=logger,
)


tft = TemporalFusionTransformer.from_dataset(
# tft = AutoRegressiveBaseModelWithCovariates.from_dataset(
    training,
    learning_rate=0.03,
    hidden_size=32,
#     attention_head_size=1,
#     dropout=0.1,
    hidden_continuous_size=8,
#     output_size=1,  # 7 quantiles by default
#     loss=RMSE(),
    log_interval=10,  # uncomment for learning rate finder and otherwise, e.g. to 10 for logging every 10 batches
    reduce_on_plateau_patience=4,
)
```

**Comment by leon90dm** [comment]
Solved by using EncoderNormalizer as target_normalizer.
